src/server.js

```diff
@@ -0,0 +1,94 @@
+import express from "express";
+import dotenv from "dotenv";
+import cors from "cors";
+import mongoose from "mongoose";
+
+import booksData from "../data/books.json";
+import { Book } from "./models/Book";
+
+dotenv.config();
+
+const mongoUrl = process.env.MONGO_URL;
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+const port = process.env.PORT;
+const app = express();
+
+app.use(cors());
+app.use(express.json());
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState !== 1) {
+    res.status(500).json({ message: "Internal server error" });
+  } else {
+    next();
+  }
+});
+
+app.get("/seed", (req, res) => {
+  try {
+    const seedDB = async () => {
+      await Book.deleteMany();
+      booksData.forEach((book) => new Book(book).save());
+    };
+    seedDB();
+    res.json({ message: "seeding completed!" });
+  } catch {
+    res.status(500).json({ message: "something went wrong!" });
+  }
+});
+
+app.get("/", (req, res) => {
+  res.send("Books API. Please go to /seed to initial database");
+});
+
+app.get("/books", (req, res) => {
+  Book.find().then((data) => {
+    res.json(data);
+  });
+});
+
+app.get("/books/:id", (req, res) => {
+  try {
+    const { id } = req.params;
+    if (Number.isNaN(Number(id))) {
+      throw new Error("id should be number");
+    }
+    Book.find({ bookID: id })
+      .then((book) => {
+        res.json({ book });
+      })
+      .catch(() => {
+        throw new Error("something went wrong");
+      });
```

Comment:
Would be great to see different error messages here, like "The ID you provided was not found" or something 😉